tests/tasks/test_shell.py

```diff
@@ -53,31 +59,36 @@ def test_shell_inherits_env():
     os.environ["MYTESTVAR"] = "42"
     out = f.run()
     assert out.is_successful()
-    assert out.result[task].result == b"42"
+    assert out.result[task].result == "42"
 
 
 def test_shell_task_accepts_env():
     with Flow(name="test") as f:
         task = ShellTask()(command="echo -n $MYTESTVAR", env=dict(MYTESTVAR="test"))
     out = f.run()
     assert out.is_successful()
-    assert out.result[task].result == b"test"
+    assert out.result[task].result == "test"
 
 
 def test_shell_task_env_can_be_set_at_init():
     with Flow(name="test") as f:
         task = ShellTask(env=dict(MYTESTVAR="test"))(command="echo -n $MYTESTVAR")
     out = f.run()
     assert out.is_successful()
-    assert out.result[task].result == b"test"
+    assert out.result[task].result == "test"
 
 
-def test_shell_returns_stderr_as_well():
+def test_shell_logs_error_on_non_zero_exit(caplog):
     with Flow(name="test") as f:
-        task = ShellTask()(command="ls surely_a_dir_that_doesnt_exist || exit 0")
+        task = ShellTask()(command="ls surely_a_dir_that_doesnt_exist")
     out = f.run()
-    assert out.is_successful()
-    assert "No such file or directory" in out.result[task].result.decode()
+    assert out.is_failed()
+    print(caplog.text)
+    assert "ERROR    prefect.Task: ShellTask:shell.py" in caplog.text
+    assert (
+        " Command failed with exit code 1: ls: surely_a_dir_that_doesnt_exist: No such file or directory"
```

Comment:
The exit code here is different than in CircleCI and good be different on other machines as well; instead of hard-coding the exit code I'd recommend updating the assert to either a regex or something milder like
```python
assert "ls: surely_a_dir_that_doesnt_exist: No such file or directory" in caplog.text
```

Reply:
change made via push